docs: PCP-4359: CAPI Core Version Upgrade (#6501)
spectrocloud/librarium: docs/docs-content/clusters/data-center/nutanix/register-nutanix-cloud.md

  compatibility matrix for more information.

- A Palette account with system console access. The user with this privilege is the
  [system administrator](../../../glossary-all.md#system-administrator) of the self-hosted
  [Palette](https://docs.spectrocloud.com/enterprise-version/system-management/#system-console) or
  [VerteX](https://docs.spectrocloud.com/vertex/system-management/#system-console) instance.

- A Nutanix logo downloaded. Review logo requirements in [Register the Cloud](#register-the-cloud).

- [curl](https://curl.se/docs/install.html) command installed or the method of your choice to make API calls for Palette
  and VerteX.

<!-- - A valid Palette authentication token. To learn how to acquire an authentication token, review the [Authorization Token](https://docs.spectrocloud.com/user-management/authentication/authorization-token) guide. -->

## Customize YAML Configuration Files

Before you can register your Nutanix cloud with Palette, you must download the appropriate CAPX manifests and edit them
accordingly so that the APIs can communicate with Palette.

> Certain components are required, while others are optional. By default, Palette uses CAPI version `1.5.3`. If you want

Reviewer: ```suggestion
You need to configure certain components explicitly, while others are optional and will fall back to default settings if not configured. By default, Palette uses CAPI version `1.5.3`. If you want
```
maybe?

Author: This one is soooooo much better <3

> to upgrade CAPI to a different version, you must use all three optional components, and all three should be for the same

Reviewer: Just to confirm: If I want to upgrade only the core component, this is not allowed. That's it? :)

CAPI version.

| **Component**               | **Requirement** |
| --------------------------- | --------------- |
| `cluster-template`          | Required        |
| `controlplane-template`     | Required        |
| `infrastructure-components` | Required        |
| `worker-template`           | Required        |
| `bootstrap-components`      | Optional        |
| `controlplane-components`   | Optional        |
| `core-component`            | Optional        |

### Required Components

The following components are required to register your Nutanix cloud with Palette. Use the following procedure to
download and format the manifests appropriately.

1.  Review the
    [Nutanix compatibility matrix](https://opendocs.nutanix.com/capx/latest/validated_integrations/#validated-versions)
    to ensure your desired CAPX version is compatible with your CAPI version. Once you have verified they are
    compatible, export your CAPX version as an environment variable. For example, if you want to download version
    `v1.6.0`, issue the following command.

    ```bash

>     export CAPX_VERSION="v1.6.0"

Reviewer: In the comp matrix, i dont see CAPX version 1.6.0. Is that correct? Same comment for CAPI 1.9.4. Do we need to use the validated versions?

            ```bash
            kubeletExtraArgs:
              rotate-server-certificates: "true"
            ```

      </details>

If you are not upgrading the default CAPI version, proceed to the [Register the Cloud](#register-the-cloud) section of
this guide.

### Optional Components

To use a non-default CAPI version, you must download additional CAPI manifests. Take the following steps to modify your
CAPI version.

1. Export your CAPI version as an environment variable. For example, if you want to download version `v1.9.4`, issue the
   following command.

   ```bash
   export CAPI_VERSION="v1.9.4"
   ```

2. Issue the commands below to download the required components.

   ```bash
   curl --remote-name --location https://github.com/kubernetes-sigs/cluster-api/releases/download/$CAPI_VERSION/bootstrap-components.yaml
   curl --remote-name --location https://github.com/kubernetes-sigs/cluster-api/releases/download/$CAPI_VERSION/control-plane-components.yaml
   curl --remote-name --location https://github.com/kubernetes-sigs/cluster-api/releases/download/$CAPI_VERSION/core-components.yaml
   ```

> 3. Verify that you have the following files downloaded using a command such as `ls -l`.

Reviewer: After downloading the files, the user also needs to customize them according to their needs

   }
   ```

3. Copy the authorization token, assign it to the `TOKEN` variable, and export it. Replace the authorization value below
   with the value from the output.

   ```bash
   export TOKEN="**********"
   ```

4. Register the Nutanix cloud type in Palette using the `/v1/clouds/cloudTypes/register` endpoint.

   :::info

>    We recommend setting the cloud `name` as `nutanix` to automatically make the out-of-the-box `nutanix-csi` pack

Reviewer: ```suggestion
   You must set the cloud `name` as `nutanix` to automatically make the out-of-the-box `nutanix-csi` pack
```

        --form "fileName=@${infraComponents}"
   ```

> 7. (Optional) If you configured the [optional components](#optional-components) to use a specific CAPI version, register

Reviewer: Just curious, why did you choose to place this step between the cloud provider and cluster template registration? :)
